core/o11y/src/metrics.rs

```diff
@@ -183,3 +195,48 @@ pub fn try_create_histogram_vec(
     prometheus::register(Box::new(histogram.clone()))?;
     Ok(histogram)
 }
+
+static EXCEPTIONS: Lazy<HashSet<&str>> = Lazy::new(|| {
+    HashSet::from([
+        "flat_storage_cached_changes_num_items",
+        "flat_storage_cached_changes_size",
+        "flat_storage_cached_deltas",
+        "flat_storage_creation_fetched_state_items",
+        "flat_storage_creation_fetched_state_parts",
+        "flat_storage_creation_remaining_state_parts",
+        "flat_storage_creation_status",
+        "flat_storage_creation_threads_used",
+        "flat_storage_distance_to_head",
+        "flat_storage_head_height",
+    ])
+});
+
+/// Expect metrics exported by nearcore to have a common prefix. This helps in the following cases:
+/// * Avoids name conflicts with metrics from other systems.
+/// * Helps filter and query metrics
+/// * Makes it easy to understand which binary export a certain metric.
+fn check_metric_near_prefix(name: &str) -> Result<()> {
+    // Some metrics were already introduced that without the desired prefix.
+    // TODO: Consistent metric naming.
```

Comment:
Idea: link a GitHub issue in this "TODO(#XXXX)" to track work on removing the exceptions